pallets/pools/src/tranche.rs

```diff
@@ -777,8 +799,10 @@ where
 		let mut prices = self.combine_mut_non_residual_top(|tranche| {
 			let total_issuance = Tokens::total_issuance(tranche.currency);
 
-			if pool_is_zero || total_issuance == Zero::zero() {
-				Ok(One::one())
+			if total_issuance == Zero::zero() {
+				Ok(BalanceRatio::one())
+			} else if pool_is_zero {
+				Ok(BalanceRatio::zero())
```

Comment:
If a pool is zero after passing the issuance is not zero clause, this means we have lost all assets on a pool that had value. Hence, we should make the tokens-price zero.